src/gui/src/drcWidget.cpp

```diff
@@ -765,27 +765,54 @@ void DRCWidget::loadJSONReport(const QString& filename)
                    filename.toStdString(),
                    e1.what());
   }
-
+  auto tech = block_->getDataBase()->getTech();
+  int dbUnits = block_->getDataBase()->getTech()->getDbUnitsPerMicron();
+  // check if DRC key exists
+  if (tree.find("DRC") == tree.not_found()) {
+    logger_->error(utl::GUI,
+                   87,
+                   "Unable to find the DRC key in JSON file {}",
+                   filename.toStdString());
+  }
   for (const auto& rule : tree.get_child("DRC")) {
     auto& drc_rule = rule.second;
 
-    const std::string violation_type = drc_rule.get<std::string>("name");
-    const std::string violation_text = drc_rule.get<std::string>("description");
-
+    const std::string violation_type = drc_rule.get<std::string>("name", "");
+    const std::string violation_text
+        = drc_rule.get<std::string>("description", "");
     int i = 0;
-    for (const auto& violation_shape : drc_rule.get_child("violations")) {
-      auto& shape = violation_shape.second;
+    auto violations_arr = drc_rule.get_child_optional("violations");
+    if (!violations_arr) {
+      logger_->error(utl::GUI,
+                     86,
+                     "Unable to find the violations key in JSON file {}",
+                     filename.toStdString());
+    }
+    for (const auto& [_, violation] : violations_arr.value()) {
+      std::string layer_str = violation.get<std::string>("layer", "-");
+      const std::string shape_type = violation.get<std::string>("type", "-");
+      odb::dbTechLayer* layer = nullptr;
+      if (!layer_str.empty()) {
+        layer = tech->findLayer(layer_str.c_str());
+        if (layer == nullptr) {
+          logger_->warn(
+              utl::GUI, 79, "Unable to find tech layer: {}", layer_str);
+        }
+      }
 
       std::vector<odb::Point> shape_points;
-      for (const auto& shape_pt : shape.get_child("shape")) {
-        auto& pt = shape_pt.second;
-        shape_points.emplace_back(
-            pt.get<double>("x") * block_->getDbUnitsPerMicron(),
-            pt.get<double>("y") * block_->getDbUnitsPerMicron());
+      auto shape = violation.get_child_optional("shape");
```

Comment:
Does this imply that there can only be one shape per violation?  Previously it was a for loop.

Reply:
the for loop is still there. line 806